t/22-svirt.t

```diff
@@ -203,4 +203,73 @@ subtest 'SSH usage in svirt' => sub {
+    my $module = Test::MockModule->new('backend::baseclass');
+    my @LAST_;
+    my $test_log_cnt = 0;
+    my $grep_return  = 1;
+    my @deleted_logs;
+    $module->mock('run_ssh_cmd', sub {
+            my $self = shift;
+            @LAST_ = @_;
+            my $cmd = shift;
+            return !!($test_log_cnt > 0 ? --$test_log_cnt : 0) if ($cmd =~ m/^test -e/);
+            return $grep_return if ($cmd =~ m/^grep -q/);
+            push @deleted_logs, ($cmd =~ /(\S+)$/) if ($cmd =~ / && rm /);
+            return (0, "FOOBAR_OUTPUT", '') if ($cmd =~ m/^cat /);
+            die("Adopt test, unexpected call of run_ssh_cmd()");
+    });
+
+    my $run_ssh_expect = '$a';
+    $module->mock('run_ssh', sub {
+            my ($self, $cmd, %args) = @_;
+            like($cmd, qr/$run_ssh_expect/, "Expect '$cmd' like qr/$run_ssh_expect/");
+            return ('A', 'B');
+    });
+
+    delete $bmwqemu::vars{VIRSH_VMM_FAMILY};
+    $bmwqemu::vars{JOBTOKEN} = 'XYZ23';
+
+    my $svirt = backend::svirt->new();
+    $run_ssh_expect = 'virsh console NAME\s+;';
+    $svirt->open_serial_console_via_ssh('NAME');
+
+    $run_ssh_expect = 'virsh console NAME DEV\s*;';
+    $svirt->open_serial_console_via_ssh('NAME', devname => 'DEV');
+    $run_ssh_expect = 'virsh console NAME DEV666\s*;';
+    $svirt->open_serial_console_via_ssh('NAME', devname => 'DEV', port => 666);
+    $run_ssh_expect = 'virsh console NAME 666\s*;';
+    $svirt->open_serial_console_via_ssh('NAME', port => 666);
+
+    $bmwqemu::vars{VIRSH_VMM_FAMILY} = 'vmware';
+    $bmwqemu::vars{VMWARE_HOST}      = 'my.vmware.host';
+    $run_ssh_expect                  = 'nc my.vmware.host\s*;';
+    $svirt->open_serial_console_via_ssh('NAME');
+    $run_ssh_expect = 'nc my.vmware.host 666\s*;';
+    $svirt->open_serial_console_via_ssh('NAME', port => 666);
+
+    $bmwqemu::vars{VIRSH_VMM_FAMILY} = 'hyperv';
+    $bmwqemu::vars{HYPERV_SERVER}    = 'my.hyperv.server';
+    $run_ssh_expect                  = 'nc my.hyperv.server\s*;';
+    $svirt->open_serial_console_via_ssh('NAME');
+    $run_ssh_expect = 'nc my.hyperv.server 666\s*;';
+    $svirt->open_serial_console_via_ssh('NAME', port => 666);
+
+    # Disable command check, as we do not care anymore
+    $module->mock('run_ssh', sub { return ('A', 'B') });
+    is_deeply([$svirt->open_serial_console_via_ssh('NAME')], ['A', 'B'], 'Check that we get output from run_ssh() call');
+
+    $bmwqemu::vars{JOBTOKEN} = 'CHECK_DELETE_TOKEN';
+    $test_log_cnt = 11;
+    dies_ok(sub { $svirt->open_serial_console_via_ssh('NAME') }, "die() when log file wasn't created");
+
+    $test_log_cnt = 0;
+    $grep_return  = 0;
+    dies_ok(sub { $svirt->open_serial_console_via_ssh('NAME') }, 'die() when emulate CONSOLE_EXIT token in log file');
+
+    my $expected_serial_file = '/tmp/' . $svirt->SERIAL_TERMINAL_LOG_PATH . '.CHECK_DELETE_TOKEN';
+    is($_, $expected_serial_file, "Check if file get cleaned") for (@deleted_logs);
```

Comment:
When trying this out manually I found that `@deleted_logs` includes the same file name twice. I stumbled over this as I tried to make the output string unique for each step with

```perl
-    is($_, $expected_serial_file, "Check if file get cleaned") for (@deleted_logs);
+    is($_, $expected_serial_file, "Check if file '$_' gets cleaned") for (@deleted_logs);
```

would be good if you can take a look into this and potentially include my change if you agree.

Reply:
Yes, the name is there twice, that's why I made this `foreach` loop.
I think I'm going to remove this `foreach` and check `@deleted_logs` after each `open_serial_console_via_ssh()` call.

The thing is, that actually this function should be called only once, but there is
no check inside of it :/ (only a hint in perlpod https://github.com/os-autoinst/os-autoinst/blob/master/backend/svirt.pm#L261). Maybe something we need to improve anyway, but I would do it not with this PR.